docs/design/storagenode-graceful-exit-overview.md

```diff
@@ -0,0 +1,164 @@
+
+There are scenarios that this design document does not handle:
+
+- Storage node runs out of bandwidth during exiting.
+- Storage node may also want to partially exit due to decreased available capacity.
+- Storage node wants to rejoin a satellite it previously exited.
+
+## Design
+
+The design is divided into four parts:
+
+- [Process for gathering pieces that need to be transferred.](storagenode-graceful-exit-pieces.md)
+- [Protocol for transferring pieces from one storage node to another.](storagenode-graceful-exit-protocol.md)
+- [Reporting for graceful exit process.](storagenode-graceful-exit-reports.md)
+- [User Interface for interacting with graceful exit.](storagenode-graceful-exit-ui.md)
+
+Overall a good graceful exit process looks like:
+
+1. Storage node Operator initiaties the graceful exit process, which:
+    - notifies the satellite about the graceful exit, and
+    - adds entry about exiting to storagenode database.
+2. Satellite receives graceful exit request, which:
+    - adds entry about exiting to satellite database, and
+    - starts gathering of pieces that need to be transferred.
+3. Satellite finishes gathering pieces that need to be transferred.
+4. Storage node keeps polling satellite for pieces to transfer.
+5. When the satellite doesn't have any more pieces to transfer, it will respond with a completion receipt.
+6. Storage node stores completion information in the database.
+7. Satellite Operator creates a report for exited storage nodes in order to release escrows.
+
+For all of these steps we need to ensure that we have sufficient monitoring.
+
+When a Graceful Exit has been started, it must either succeed or fail. The escrow, held from storage node, will be released only on success. We will call the failure scenario an ungraceful exit.
+
+Ungraceful exit can happen when:
+
+- Storage node doesn't transfer pieces,
+- Storage node incorrectly transfers pieces,
+- Storage node is too slow to transfer pieces,
+- Storage node decided to terminate the process.
+
+## Implementation
+
+To coordinate the four parts we need few things implemented:
+
+- Add `satellites` table and interfaces to storage node.
+- Add `satellites_exit_progress` table and interfaces to storage node.
+- Update `nodes` table on satellite.
+- Add `nodes_exit_progress` table to satellite.
+
+### Storage Node Database Changes
+
+Create `satellites` table:
+
+```
+model satellites (
+    key node_id
+
+    field node_id  blob not null
+    field address  text not null
+    field added_at timestamp ( autoinsert ) not null
+
+    field status   byte not null
+)
+```
+
+Create `satellites_exit_progress` tables:
+
+```
+model satellites_exit_progress (
+    fk satellite_id 
+
+    field initiated_at         timestamp ( updateable )
+    field finished_at          timestamp ( updateable )
+    field starting_disk_usage  int64 not null
+    field bytes_deleted        int64 not null
+    field completion_receipt   blob
+)
+```
+
+### Rationale
```

Comment:
Rationale should be second level heading and let's move it after satellite database changes